fun/Animal.cc.lua

```diff
@@ -0,0 +1,51 @@
+{{/* This is an API command, where you can generate pictures of ducks, foxes, goats, cats, shibas, and my personal favourite, http Cats. I will probably add more afterwards, but this is it for now
+     Recommended Trigger type: Command trigger, Trigger: animal.
+     Please note that the animal names are case sensitive, the first letter of the animal is capital. If you have any animal pictures or APIs, feel free to DM me (GenryMg#0001) I'll add them here. 
+*/}}
+
+{{$args := parseArgs 0 "Usage: -animal <animal type>"
+(carg "string" "animaltype")
+(carg "string" "http")}}
+{{$animal := ($args.Get 0)}}
+
+{{if eq $animal "fox"}}
+{{$link := "https://randomfox.ca/images/"}}
+{{$num := randInt 1 122}}
+{{$msglink := joinStr "" $link $num ".jpg" }}
+{{sendMessage nil (cembed "title" "Have a Fox!" "image" (sdict "url" $msglink))}}
+{{/* The next part is the same as satty's randduck commad. I just copy and pasted :P */}}
+{{else if eq $animal "duck"}}
+{{ $link := "https://random-d.uk/api/" }}
+{{ $c := randInt 10 }}
+{{ if lt $c 7 }}
+{{ $link = joinStr "" $link (toString (randInt 1 130) ) ".jpg" }}
+{{ else }}
+{{ $link = joinStr "" $link (toString (randInt 1 29)) ".gif" }}
+{{ end }}
+{{ sendMessage nil (cembed "title" "Have a duck!" "image" (sdict "url" $link)) }}
+{{else if eq $animal "goat"}}
+{{$link := "https://placegoat.com/600/600"}}
+{{sendMessage nil (cembed "title" "Have a Goat!" "image" (sdict "url" $link))}}
+{{else if eq $animal "httpcat"}}
+{{if gt (len .Args) 2}}
+{{$link := "https://http.cat"}}
+{{$https := ($args.Get 1)}}
+{{$link = (joinStr "/" $link $https)}}
+{{ sendMessage nil (cembed "title" "Have a http cat!" "image" (sdict "url" $link)) }}
+{{else}}
+{{sendMessage nil "Please provide an http response"}}
+{{end}}
+{{else if eq $animal "cat"}}
+{{$link := joinStr "" "https://cataas.com/cat?filter=" (urlescape adjective)}}
+{{$ID := sendMessageRetID nil (cembed "image" (sdict "url" $link))}}
+{{sleep 2}}
+{{$msg := getMessage nil $ID}}
+{{editMessage nil $ID (cembed "title" "Have a cat!" "image" (sdict "url" (index $msg.Embeds 0).Image.ProxyURL))}}
```

Comment:
Why is there a need for this edit? Seems to work fine for me without.

Reply:
uhhh. Satty wrote that one some time ago, I don't know actually.

Reply:
I asked him, and he said it's becuase of discord not caching correctly, causing formatting problems and problems with the output. I tested it without the sleep, and sometimes it sneds one cat but then displays another when you click on it.